### ISSUE TITLE
[Airbrake] [Production] SMTP To address may not be blank: []

### ISSUE THREAD

**Issue by aspiers** [swapmyvote/swapmyvote#249]
**Airbrake error:** [#8955](https://herokuapp32817511herokucom.airbrake.io/projects/110137/groups/2625620460085418955)
**Airbrake project:** swapmyvote

**Error type:** `ArgumentError`
**Error message:** `` SMTP To address may not be blank: [] ``
**Where:** `swaps#create`
**Occurred at:** Dec 02, 2019 15:59:08 UTC
**First seen at:** Dec 02, 2019 15:59:08 UTC
**Occurrences:** 0 (0 since last deploy on Dec 02, 2019 14:05:27 UTC)
**Severity:** `error`

**URL:** [https://www.swapmyvote.uk/user/swap?user_id=22551](https://www.swapmyvote.uk/user/swap?user_id=22551)
**File:** `/GEM_ROOT/gems/mail-2.7.1/lib/mail/check_delivery_params.rb`

**Backtrace:**
```
/GEM_ROOT/gems/mail-2.7.1/lib/mail/check_delivery_params.rb:21:in check_to
/GEM_ROOT/gems/mail-2.7.1/lib/mail/check_delivery_params.rb:7:in check
/GEM_ROOT/gems/mail-2.7.1/lib/mail/network/delivery_methods/smtp_connection.rb:52:in deliver!
/GEM_ROOT/gems/mail-2.7.1/lib/mail/network/delivery_methods/smtp.rb:101:in block in deliver!
```

**Comment by aspiers** [comment]
The only thing we can do if we don't have the recipient address is to skip sending the email.

**Comment by aspiers** [comment]
They can have no email set if they denied permission for Facebook or Twitter to share their email, or if they managed to blank the address from their SMV account or from their Facebook or Twitter account (not sure which of those are possible).

**Comment by jpallen** [comment]
We used to insist on email, and prompt them to add it if we didn't get it from Twitter/Facebook - has this changed?

**Comment by jpallen** [comment]
I think we need to insist on email here, and not offer to swap with people without an email, because otherwise they will never know about it, and that's not fair to the person proposing the swap for it just fail silently.

**Comment by baob** [comment]
I agree with @jpallen ... If we can't get a message to the 'other' then maybe we shouldn't be offering them to the 'proposer' ... they could be filtered out [here](https://github.com/swapmyvote/swapmyvote/blob/master/app/models/user.rb#L35)

**Comment by aspiers** [comment]
I don't ever remember seeing that insistence but if it was there I'm pretty sure it was never *deliberately* removed.  I'm 100% on board with insisting on it, however in this case #15 becomes more important.

**Comment by aspiers** [comment]
> I think we need to insist on email here, and not offer to swap with people without an email, because otherwise they will never know about it

Well they will know about it if they revisit the website.  But I agree it's not ideal and insisting on email would be better.

**Comment by aspiers** [comment]
Fixed by #253.
